Pod/Classes/SEGAdobeIntegration.m

```diff
@@ -425,6 +397,23 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
         SEGLog(@"[ADBMediaHeartbeat trackSessionEnd]");
         return;
     }
+
+    NSDictionary *videoTrackEvents = @{
+        @"Video Playback Buffer Started" : @(ADBMediaHeartbeatEventBufferStart),
+        @"Video Playback Buffer Completed" : @(ADBMediaHeartbeatEventBufferComplete),
+        @"Video Playback Seek Started" : @(ADBMediaHeartbeatEventSeekStart),
+        @"Video Playback Seek Completed" : @(ADBMediaHeartbeatEventSeekComplete)
+    };
+
+    for (NSString *event in videoTrackEvents) {
```

Comment:
Iterating over the entire dictionary is inefficient. We should remove it - the check should just check if the event exists in the `videoTrackEvents` dictionary or not.

Reply:
Hmm but I want to dynamically set it with the Adobe value

Reply:
you can still do that.

```
if (videoTrackEvents[event]) {
  enum ADBMediaHeartbeatEvent videoEvent = [videoTrackEvents[event] intValue];
  ...
}
```

Or even:
```
enum ADBMediaHeartbeatEvent videoEvent = [videoTrackEvents[event] intValue];
if videoEvent != 0 {
  ...
}

```